jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/job/Shipment.java

```diff
@@ -448,6 +448,21 @@ public TimeWindow getPickupTimeWindow() {
         return pickupTimeWindows.getTimeWindows();
     }
 
+    
+    /**
+     * Returns a string with the shipment's attributes.
+     * <p>
+     * <p>String is built as follows: [attr1=val1][attr2=val2]...
+     */
+    @Override
+    public String toString() {
+        return "[id=" + id + "][name=" + name + "][pickupLocation=" + pickupLocation_
+                + "][deliveryLocation=" + deliveryLocation_ + "][capacity=" + capacity
+                + "][pickupServiceTime=" + pickupServiceTime + "][deliveryServiceTime="
+                + deliveryServiceTime + "][pickupTimeWindow=" + pickupTimeWindow
+                + "][deliveryTimeWindow=" + deliveryTimeWindow + "]";
+    }
+
```

Comment:
@michalmac Thanks. There might be more than one pickup and delivery time window. Would you mind to account for this also? For example, [pickupTimeWindow1=][pickupTimeWindow2=] or [pickupTimeWindow_1 ...]?

Reply:
or like this "[pickupTimeWindows=[timeWindow=" + pickupTimeWindow[0] + "][...]]“

Reply:
Sure, my change was according to what was in Service. But now also Service.toString() prints out all time windows.

BTW. I cleaned up slightly both classes. There are still 2 things that could be done:
- get rid of timeWindow from ServiceBuilder and pickup/deliveryTimeWindow from ShipmentBuilder -- they are unused
- we could get rid of some generics-related warnings by changing XXX.Builder.newInstance() to XXX.Builder.newXXXInstance in Service/Break/Delivery/Pickup... but this is a change to API, so not sure if we want it